skills/singlecell/scrna/sc-batch-integration/sc_integrate.py

```
@@ -709,6 +725,15 @@ def main():
         adata.raw = adata.copy()
         sc.pp.highly_variable_genes(adata, n_top_genes=2000, layer="counts", flavor="seurat_v3")
         sc.pp.pca(adata)
+        try:
+            processed_demo, _ = sc_io.load_repo_demo_data("pbmc3k_processed")
+            label_key = "louvain" if "louvain" in processed_demo.obs.columns else "leiden" if "leiden" in processed_demo.obs.columns else None
+            if label_key:
+                aligned = processed_demo.obs.reindex(adata.obs_names)[label_key]
+                if aligned.notna().any():
+                    adata.obs[label_key] = aligned.astype(str)
+        except Exception as exc:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)